databricks/koalas/config.py

```diff
@@ -126,14 +138,18 @@ def reset_option(key: str) -> None:
     default_session().conf.unset(_key_format(key))
 
 
-def _check_option(key: str, value: Union[str, _NoValueType] = _NoValue) -> None:
+def _check_option(key: str) -> None:
     if key not in _registered_options:
         raise OptionError(
             "No such option: '{}'. Available options are [{}]".format(
                 key, ", ".join(list(_registered_options.keys()))))
 
-    if value is None:
-        return  # None is allowed for all types.
-    if value is not _NoValue and not isinstance(value, type(_registered_options[key])):
-        raise TypeError("The configuration value for '%s' was %s; however, %s is expected." % (
-            key, type(value), type(_registered_options[key])))
+
+def _convert_value(key: str, value: Any) -> Any:
+    convert_func = _registered_options[key][1]
+    if value is not _NoValue:
```

Comment:
maybe still need to give an option if `value is None`?

Reply:
right now, it seems to me for instance, if `config.test.something: (100, int)`, and then someone sets this option to `None`, and this `None` is allowed value for this option, i guess there will be an error raised?

Reply:
Yup, now it will throw an error. However, `None` can be also allowed explicitly such as https://github.com/databricks/koalas/pull/739/files#diff-4e0dcdf52811413fc3041eb7a9ef6c10R31

Currently, `None` is not allowed for the options we support currently.

Reply:
ok, so for options that allow `None`, you mean something like `plotting.max_rows: (1000, (int, None))` will work then?

Reply:
Ah yup. I was thinking that's better.